### ISSUE TITLE
tests for dots broken for python 3.8 3.9 with "last lambda bin is not empty"

### ISSUE THREAD

**Issue by sergpolly** [open2c/cooltools#392]
this is all due to https://github.com/pandas-dev/pandas/issues/48645

and is already fixed in `pandas 1.6.dev`

that's where is breaks with pandas `1.5.0`: https://github.com/open2c/cooltools/blob/a5341aa03f1bbcc1087983f2919602d4f25c333a/cooltools/api/dotfinder.py#L897

if curious check minimal reproducible example:
```python
import numpy as np
import pandas as pd

data = pd.DataFrame({"a":[0,0,0,1,1,2]})
bins  = np.r_[-np.inf,[1,2,3,4],np.inf]
categories = pd.cut(data["a"],bins)
# outputs below should be identical - but they are not with pandas 1.5.0
display(data.groupby(categories,dropna=True).count())
display(data.groupby(categories,dropna=False).count())
```

**Comment by Phlya** [comment]
I think the freshly released pandas 1.5.1 has this bugfix included

**Comment by gfudenberg** [comment]
with
```python
a = data.groupby(categories,dropna=True).count()
b = data.groupby(categories,dropna=False).count()
a.equals(b)```
in 1.5.1 !
